client/src/index.ts

```diff
@@ -470,3 +479,30 @@ export class XataError extends Error {
 }
 
 export type Links = Record<string, Array<string[]>>;
+
+const isBranchStrategyBuilder = (strategy: BranchStrategy): strategy is BranchStrategyBuilder => {
+  return typeof strategy === 'function';
+};
+
+const isBranchStrategy = (strategy: BranchStrategy): strategy is BranchStrategy => {
+  return isBranchStrategyBuilder(strategy) || typeof strategy === 'string';
+};
+
+const getBranchFromStrategy = async (param: BranchStrategy | BranchStrategy[]): Promise<string | undefined | null> => {
+  const strategies = Array.isArray(param) ? param : [param];
+
+  const getBranch = async (strategy: BranchStrategy) => {
+    if (isBranchStrategyBuilder(strategy)) {
+      return await strategy();
+    } else if (isBranchStrategy(strategy)) {
+      return strategy;
+    } else {
+      throw new Error(`Invalid branch strategy: ${strategy}`);
+    }
+  };
+
+  const strategy = strategies.find((option) => getBranch(option) !== undefined);
+  if (strategy) return await getBranch(strategy);
```

Comment:
I tried to make the branch strategy look-up lazy and stop on the first not-undefined value, however we are calling it twice now. Although the actions here should never be destructive, they could be slow. Not sure if it's worth to refactor it to a reduce that stops early but does not call it twice (less readable) or mutating a local variable in a loop (I personally try to avoid them as code style, but I'd like your feedback on how you normally approach these implementation details).